apps/examples/testcase/le_tc/kernel/itc_pthread.c

```diff
@@ -0,0 +1,538 @@
+
+#include <pthread.h>
+#include <stdio.h>
+#include <errno.h>
+#include <pthread.h>
+#include <stdlib.h>
+#include <tinyara/pthread.h>
+#include <tinyara/config.h>
+#include "tc_internal.h"
+#include <sys/types.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <signal.h>
+#include <sched.h>
+
+#define WAIT_TIME_1                   1
+#define WAIT_TIME_2                   2
+#define WAIT_TIME_8                   8
+#define RETURN_PTHREAD_JOIN     ((void *)100)
+#define BUF_LEN 32
+#define LOOP_SIZE CONFIG_ITC_KERN_PTHREAD_LOOP_SIZE
+
+/****************************************************************************
+ * Helper Functions
+ **************************************************************************/
+static pthread_t g_pthread;
+static bool g_flag;
+static pthread_mutex_t g_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_key_t g_key;
+static int  g_exec_index;
+
+/**
+* @fn                   :thread_setgetname
+* @brief                :utility function for creating thread
+* @return               :void*
+*/
+static void *thread_setgetname(void *param)
+{
+	while (1) {
+		sleep(WAIT_TIME_1);
+		if (g_exec_index == LOOP_SIZE)
+			break;
+	}
+	return NULL;
+}
+/**
+ *@fn					:thread_setget
+ *@breif				:utility function for creating thread
+ *@return               :void*
+ */
+static void *thread_setget(void *param)
+{
+	sleep(WAIT_TIME_2);
+	return NULL;
+}
+
+
+
+/**
+* @fn                   :thread_equal_check
+* @brief                :utility function for itc_pthread_pthread_equal_p
+* @return               :void*
+*/
+static void *thread_equal_check(void *param)
+{
+	if (pthread_equal(g_pthread, pthread_self()) == false) {
+		g_flag = false;
+		return NULL;
+	}
+	g_flag = true;
+	return NULL;
+}
+
+/**
+* @fn                   :thread_set_get_specific
+* @brief                :utility function for itc_pthread_key_create_set_getspecific_p_multitime
+* @return               :void*
+*/
+static void *thread_set_get_specific(void *param)
+{
+	int *get_spec;
+	int ret_chk;
+	int exec_index;
+	g_flag = false;
+
+	for (exec_index = 1; exec_index <= LOOP_SIZE; exec_index++) {
+		ret_chk = pthread_setspecific(g_key, &exec_index);
+		if (ret_chk != OK) {
+			printf("itc_pthread_key_create_set_getspecific_p_multitime: TC FAIL\n");
+			return NULL;
+		}
+
+		get_spec = pthread_getspecific(g_key);
+		if (*get_spec != exec_index) {
+			printf("itc_pthread_key_create_set_getspecific_p_multitime: TC FAIL\n");
+			return NULL;
+		}
+	}
+	sleep(WAIT_TIME_1);
+	g_flag = true;
+	return NULL;
+}
+
+/**
+* @fn                   :task_thread
+* @brief                :utility function for itc_pthread_pthread_kill_n_rekill
+* @return               :void*
+*/
+static void *task_thread(void *param)
+{
+	sigset_t mask;
+	siginfo_t   info;
+
+	sigemptyset(&mask);
+	sigaddset(&mask, SIGUSR1);
+	pthread_sigmask(SIG_BLOCK, &mask, NULL);
+
+	sigwaitinfo(&mask, &info);
+	printf("Signal received\n");
+
+	pthread_exit(RETURN_PTHREAD_JOIN);
+
+	return NULL;
```

Comment:
@bipin4321 
Please check for valid return(I think this is not reachable code as it exits before reaching this)

Reply:
This comment has been fixed.